BackEndApp/authv1/views.py

```diff
@@ -93,3 +94,65 @@ def logout(request):
         status = 500
 
     return JsonResponse({"message": message}, status=status)
+
+
+@api_view(["POST"])
+def ForgotPassword(request):
+    try:
+        username = request.data.get("username")
+        user = User(username=username, password=None)
+        user = user.find()
+
+        if user is None:
+            status = 401
+            message = "User does not exist"
+
+        email = user["email"]
+        subject = "Auto-DL Request for Reset password"
+        msg = (
+            "You ("
+            + username
+            + ") requested a password reset. Click here to reset your password http://127.0.0.1:8000/auth/password-reset/"
+        )
+        send_mail(subject, msg, EMAIL_HOST_USER, [email])
+
+        message = "Email Sent!!"
+        status = 200
+
+    except Exception as e:
+        message = "Some error occurred!! Please try again."
+        status = 500
+
+    return JsonResponse({"message": message}, status=status)
+
+
+@api_view(["POST"])
+def PasswordReset(request):
+    try:
+        username = request.data.get("username")
+        user = User(username=username, password=None)
+        this_user = user.find()
+
+        if this_user is None:
+            status = 401
+            message = "User does not exist"
+
+        new_password = request.data.get("password")
+        hashed_password = bcrypt.hashpw(new_password.encode("utf-8"), bcrypt.gensalt())
+        old_password = this_user["password"]
```

Comment:
```suggestion
        old_password = this_user.get("password", "")
```
Yes, a password would exist in all cases, but let's not break here in case it doesn't.

Reply:
Yes Thanks...will do that:)